### ISSUE TITLE
Kasli v1.0 tests & errata

### ISSUE THREAD

**Issue by marmeladapk** [m-labs/sinara#349]
To be changed for rev1.1

- Misc:
  - [x] R57 must be 300R
  - [x] Improve signal integrity for backplane LVDS lines (remove reference plane cut) (one cut between 2v5 and 12V0 seems unavoidable)
  - [x] Add two more USER_LEDs
  - [x] (**v1.1/rj**) consider adding a PCB-mount heatsink (one of the FPGAs died from overheating 
during testing)
  - [ ] Document power supply design parameters (#317)
  - [x] update 
  - [x] and double check power budget after changes.
  - [x] add a PN for that clock balun on the schematic
  - [ ] Change revision to 1.1 on schematics
- USB:
  - [x] (**v1.1/rj**) Swap I2C with UART on FTDI chip (channel C doesn't have MPSSE)
  - [x] (**v1.1/rj**) fix direction of the IC19 (both channels)
- I2C:
  - [x] (**v1.1/rj**) add TVS on EEM I2C
  - [x] (**v1.1/rj**) add ID chip 24AA02E48T-I/OT, same I2C port as the Si5324
  - [x] (**v1.1/rj**) Connect reset of the I2C routers IC14 IC15 to the FPGA
  - [x] Add TCA9517 to Si5324 I2C
  - [x] Connect IC10:A1 (I2C_SW_RESET) to one of the unused BDBUS lines (BDBUS7 or BDBUS5 if you can) and **not** to CDBUS4.
- Backplane:
  - [x] (**v1.1/rj**) Inconsistent power connector PN 712RA vs L712RA (712RA is the correct one) (it was just wrong footprint name, PN was correct; PK)
  - [x] (**v1.1/rj**) Remove second power connector J17.
  - [x] (**v1.1/rj**) Populate the backplane connector J20.
- SFP:
  - [x] Move SFP LEDs closer to panel (same as was done with USB)
  - [x] Free up six pins on the FPGA by having MOD_DEF1 and MOD_DEF2 routed through the I2C switch only and not additionally directly to the FPGA.
  - [x] Remove the I2C level converters IC10 on those SFP I2C busses.
  - [x] (**v1.1/rj**) Connect RATE_SELECT1 (pin 9 of the SFP) to the FPGA
- Clock distribution
  - [x] (**v1.1/rj**) Remove the 50 MHz oscillator OSC1. Not needed as the 125 MHz GTP input can be routed to fabric.
  - [x] (**v1.1/rj**) Remove IC3
  - [x] Remove the CLK_SEL logic of IC3 (free up FPGA pin)
  - [x] (**v1.1/rj**) Remove J19 and related components
  - [x] (**v1.1/rj**) Remove J5-J7 and related components
  - [x] Remove the R98 path of CLK_OUT (IC1:Q2 to the front panel SMA)
  - [x] (**v1.1/rj**) Add an LDO for the power supply of IC1 and IC2 (also worth adding a ferrite/inductor between the SMPS and the LDO to kill high-frequency spurs).
  - [x] (**v1.1/rj**) Connect four MMCX (J1-J4) to the Q2, Q2B, Q3, Q3B outputs of IC1
  - [x] (**v1.1/rj**) Mark J2 and J4 (on the negative Q2B and Q3B outputs) clearly on the PCB
  - [x] (**v1.1/rj**) Connect backplane clock CLK_EXT_P/N to CKOUT2+- of IC2
  - [x] (**tph**) change 0R resistor to 100k to float SMA clock input
  - [x] (**tph**) consider floating MMCX connector grounds in same way as SMAs (only if you have time!)
  - [x] Change the transformer TR1 to [TC2-1TX+](https://ww2.minicircuits.com/pdfs/TC2-1TX+.pdf). No changes to layout. Footprint and package are compatible. **But pinout is rotated 180 degrees.**
  - [x] DNP C13, C9
- Layout:
  - [x] Either teardrops everywhere or nowhere. Right now there are teardrops on roughly half the pins.
  - [x] Set up consistent rules
  - [x] Properly clean up polygons
  - [ ] Don't do arcs where they are not needed.
  - [x] Remove weird teardrop on FTDI
  - [x] Connect PCB edge GND stitching vias to top and bottom GND pour. Stick with @gkasprow recommendation on via density.
  - [x] Check XADC connection to components
  - [x] Replace C49 with a 0R (connect VREFN to XADC_GND) to use internal reference (connected without jumper, PK)

**Comment by jordens** [comment]
@marmeladapk @gkasprow Is Kasli back from the board house?

**Comment by gkasprow** [comment]
Components are delivered, the PCBs were shipped yesterday. They will start assembly on Monday. It takes roughly 2 days.

**Comment by hartytp** [comment]
Wonderful! Really looking forward to seeing these!

**Comment by gkasprow** [comment]
Kasli photo:
![2017-12-15 14 41 27](https://user-images.githubusercontent.com/4325054/34049716-b9594bdc-e1b8-11e7-9e80-96301ddbcfe7.jpg)

**Comment by gkasprow** [comment]
Power supplies work well. It consumes 0.2A@12V
USB works as well. At least I see 4 serial ports under Windows.
![2017-12-15 16 52 00](https://user-images.githubusercontent.com/4325054/34049763-e438f794-e1b8-11e7-9736-1c6a112a842c.jpg)
That's all what I can do at home.

**Comment by hartytp** [comment]
Great!

**Comment by jordens** [comment]
Awesome!

**Comment by gkasprow** [comment]
Progress;
JTAG works,
FLASH programming and booting works. There is litle bug, R57must be3 00R
I loaded design which causes 2.5V rail to switch off. With little help of external PSU it wakes up but the board takes 600mA and FPGA gets hreally hot.
Then we realised that it must be split termination that is switched on on all EEM LVDS lines because there are just ordinary registers attached so far.
And indeed, the construct below instantiates uncallibrated split impedance
```

-- LVDS bidir buffers
	eem0 : for I in 0 to 7 generate
		lvds_buf : IOBUFDS_INTERMDISABLE
			GENERIC MAP (
				DIFF_TERM         => "true",
				IBUF_LOW_PWR      => "true",
				IOSTANDARD        => "LVDS_25",
				USE_IBUFDISABLE   => "false")
			PORT MAP (
				O                 => eem0_I(I),
				IO                => eem0_p(I),
				IOB               => eem0_n(I),
				INTERMDISABLE     => eem0_term_dis(I),
				I                 => eem0_O(I),
				IBUFDISABLE       => '0',
				T                 => eem0_S(I) -- 3-state enable input, high=input, low=output
			);
```

**Comment by gkasprow** [comment]
clock distribution works as well

**Comment by jordens** [comment]
Great job @gkasprow and @marmeladapk.

**Comment by gkasprow** [comment]
@jordens The only remaining thing we need to test is Silabs, I2C distribution and EEMs itself. Hopefully we will ship it in Wednesday. We can ship it as it is tomorrow and play with remaining boards that will be delivered on Wednesday.

**Comment by jordens** [comment]
No need to rush it then. I'll be gone Friday until the 28th. You can time it so that it arrives at the 28th.

**Comment by gkasprow** [comment]
I2C and silabs chip also work.

**Comment by gkasprow** [comment]
UART works as well. I talk with FORTH CPU

**Comment by marmeladapk** [comment]
@jordens I removed arcs remark, because they were added on purpose - reducing spikes of current density. Same as with you last remark about polygons (believe me, Greg bugged me about this).

We need to switch UART with I2C on FTDI chip, because channel C doesn't have MPSSE. I made bitbang I2C using pyftdi which works (I can program Si5324 and I2C switches), but its timings are horrible. 

SFP0 and SFP2 seem to work on 6,25 Gbps. There is a problem with SFP1 TX line.

![sf0-sfp2-ber6g](https://user-images.githubusercontent.com/24512604/34343686-7487c97e-e9d9-11e7-9e5a-1d66eed06133.PNG)
![sf1-sfp2-ber6g](https://user-images.githubusercontent.com/24512604/34343689-77925f26-e9d9-11e7-8208-c2a1f1c8d67c.PNG)

**Comment by marmeladapk** [comment]
@jordens Also FPGA tends to get quite hot with termination enabled so please cool it with fans to avoid damage. Also, there is some kind of a problem with XADC, sometimes it shows all voltages = 0 V and temperature = 0 K.

**Comment by hartytp** [comment]
> @jordens Also FPGA tends to get quite hot with termination enabled so please cool it with fans to avoid damage. Also, there is some kind of a problem with XADC, sometimes it shows all voltages = 0 V and temperature = 0 K.

Should we add a heat sink to the FPGA to help to avoid this? Maybe a nice PCB-mount one?

**Comment by jordens** [comment]
@marmeladapk Current density at a bend diverges with sharpness/mesh size in many Maxwell approximations. If you think you need an arc, that's most likely a simulation/algorithm artefact which does not correspond to reality. I'd be very surprised if there was a case on Kasli where an arc solves a problem that pair of 45 degree bends does not solve. Both on polygons and on traces.

Ack the UART/I2C switch. Could you post the bitbang i2c code somewhere?

Don't worry about the XADC. Did you test the DDR RAM?

On the GTP pairs you need to be a bit more careful with your design rules. You have some lines coupled closer to unrelated signals than to their differential pair partner!

**Comment by jordens** [comment]
@marmeladapk can't get jtag over usb to work. are DIR1 and DIR2 of IC19 wrong?

**Comment by jordens** [comment]
![image](https://user-images.githubusercontent.com/1338946/34443929-ca4dac04-ecca-11e7-9b30-5ef7ea32b470.png)
3.5 mil gap near IC18. Is that ok? Did they fix that in the fab?

**Comment by sbourdeauducq** [comment]
@marmeladapk @gkasprow ping

**Comment by gkasprow** [comment]
@jordens in IC17, the DIR is H so it means A -> B direction which is correct
But on IC19 channel 1 direction is B->A which is wrong.
Channel 2 direction is A -> B which is also wrong. This explains why it does not work
You can simply short pins 8, 5 and 9,4, we will fix it in the right way on remaining boards.
![obraz](https://user-images.githubusercontent.com/4325054/34513957-2715246e-f06b-11e7-8dd9-07861d962d06.png)

**Comment by jordens** [comment]
Yes, that's what I said.
Could you implement your fix and show me (photo) what you mean? Just shorting still has the same contention issue. You know better where to cut and scratch the lines.

**Comment by jordens** [comment]
I have MiSoC running (spi flash XIP, boot from spi flash, SDRAM working, uart working).

**Comment by gkasprow** [comment]
Shorting of the input with output via 220R resistors should do the job. Resistor is to limit current that will flow from FTDI to FPGA.
Spartan devices are known to react with latchup when 2.5V JTAG signal is treated with 3.3V logic output :)

**Comment by jordens** [comment]
Still unclear what you are suggesting. Desoldering the level converter? Cutting, scratching the traces on both sides and bridging over the converter?

**Comment by jordens** [comment]
I desoldered the level converter and added a short on TDO and 1k on TMS. JTAG over FTDI also works with 25 MHz. That seems to have saved nearly 500 mW of contention current. 3.5 W now.

**Comment by jordens** [comment]
No. That was the SFP transceiver. 4 W with, 3.5 W without.

**Comment by gkasprow** [comment]
This is exactly what I meant :)

**Comment by jordens** [comment]
On my board (number 2, thanks for putting the sticker on there) port 5 of IC14 has SDA (EEM1:SDA) shorted to ground.

**Comment by marmeladapk** [comment]
On board 3 EEM1 SDA is pulled to 3V3. Could you check if R157 is 2k2 and is connected to 3V3?

![img_20180108_144411](https://user-images.githubusercontent.com/24512604/34675116-d1800128-f488-11e7-94e9-73872b888e4e.jpg)

**Comment by jordens** [comment]
The 2k2 are there but they don't matter. The resistance to ground is 300 Ohm (310 in one polarity and 270 in the other). No EEM connected. Either the chip is broken or there is some conducting dust ;).

**Comment by marmeladapk** [comment]
@jordens Could you share your openocd setup?

**Comment by jordens** [comment]
kasli.cfg: 

```tcl
interface ftdi
ftdi_device_desc "Quad RS232-HS"
ftdi_vid_pid 0x0403 0x6011
ftdi_channel 0
ftdi_layout_init 0x0008 0x000b
ftdi_location 1:8

reset_config none
transport select jtag
adapter_khz 25000

source [find cpld/xilinx-xc7.cfg]
source [find cpld/jtagspi.cfg]
```

* The `ftdi_location` is only needed to distinguish multiple FT4232Hs on a machine. It needs to match the bus topology.
* To load a bitstream: `openocd -f kasli.cfg -c 'init; pld load 0 misoc_minisoc_kasli/gateware/top.bit; exit'` 
* To write bitstream and bios to flash: `openocd -f kasli.cfg -c 'init; jtagspi_init 0 /home/rj/src/conda/pkgs/bscan-spi-bitstreams-0.10.0+git-1/share/bscan-spi-bitstreams/bscan_spi_xc7a100t.bit; jtagspi_program misoc_minisoc_kasli/gateware/top.bin 0; jtagspi_program misoc_minisoc_kasli/software/bios/bios.bin 0x400000; exit'`

The spi proxy bitstreams are either in our openocd conda packages or at https://github.com/jordens/bscan_spi_bitstreams

And if you want to play I2C, I pushed my toy code to https://github.com/quartiq/kasli-i2c

**Comment by jordens** [comment]
As @sbourdeauducq reported elsewhere, Kasli Ethernet is working. On the board here ("#2") SFP1 also works fine.

**Comment by marmeladapk** [comment]
@jordens To build Kasli target I pulled latest Artiq repo and I wanted to update conda environment to it using `conda env update -f conda/artiq-dev.yaml`. However this didn't pull latest migen and misoc with changes for Kasli, so I had to do this manually to be able to build gateware. Is this expected behaviour?

Also, build failed with errors:  `Rule violation (IOSTDTYPE-1) IOStandard Type - I/O port eem0_n is Single-Ended but has an IOStandard of LVDS_25 which can only support Differential`. I used Vivado 2016.4. Is this known or is there some mistake on my side?

**Comment by jordens** [comment]
@marmeladapk I didn't know about `conda update` but `-f` is force and I don't think you can use it in this case. Just install the latest with `conda install -c m-labs/label/dev misoc migen openocd`. I don't know about the rule violation. I don't see those. We use `2017.2` or `2017.4`. The [buildbot](http://buildbot.m-labs.hk/waterfall) will generate binary packages for Kasli soon.

**Comment by marmeladapk** [comment]
I'll check with 2017.4, but generally I got those errors when in constraints eem lines were set as LVDS, but they didn't have IOBUFDS instantiated in code.

**Comment by jordens** [comment]
@marmeladapk Make sure that the misoc/migen that are being used are actually the ones installed through conda and not some packages that you `git clone`-ed yourself and `setup.py install` or `pip install`-ed. But I can see unused inputs causing that warning. Pretty sure it's benign.

**Comment by marmeladapk** [comment]
Huh on board 4 and 6 SFP1 also works. Seems like I didn't do the test properly.

I tested last link using @gkasprow SATA-SFP converter. Below is a screenshot of all links working simultaneously at 6,25 Gbps. 
<img width="1129" alt="wszystkie-6g" src="https://user-images.githubusercontent.com/24512604/34954173-bc8211ca-fa1f-11e7-81a2-c6dba40e1f96.PNG">

**Comment by jordens** [comment]
Si5324 clock distribution noise and spurs [running from on-board XO](https://github.com/quartiq/kasli-i2c/blob/6e1f137c0faa55276fb8bd6c1e24958200a9c0e5/i2c.py#L392-L402). Looking at J1, noise for the low-frequency part mostly limited by the SA. The one spur at ~449 kHz seems to be the SMPS. It wanders if I touch the SMPS components.
![image](https://user-images.githubusercontent.com/1338946/35051434-8087cc6e-fba5-11e7-98f8-4083c04025cd.png)
![image](https://user-images.githubusercontent.com/1338946/35052029-e04eafcc-fba6-11e7-9491-4bf4f2a59bb0.png)

**Comment by hartytp** [comment]
So, something like -100dBc/Hz at 100kHz offset? We expect something like -130dBc/Hz based on the datasheet. Wonder what limits that.

**Comment by jordens** [comment]
The [SA](http://www.siglentamerica.com/USA_website_2014/Documents/DataSheet/SSA3000X_DataSheet.pdf) is the limit on noise density everywhere but not on spurs.

**Comment by gkasprow** [comment]
The Silabs supply was not designed to be very low noise. We can add LDO.

**Comment by jordens** [comment]
I'd do that but also remove the clock fanout in favor of a single MMCX output form the Si5324 and then using Clocker or/and the backplane. Kasli is too dense and overloaded. The additional space would help with figuring out a good heatsinking solution, better SI, cleaner routing, higher yield, and lower manufacturing risk.

Also I would like to free up six pins on the FPGA by having MOD_DEF1 and MOD_DEF2 routed through the I2C switch only and not additionally directly to the FPGA. I am not aware of an SFP module that uses those pins for non-I2C applications.

**Comment by jordens** [comment]
But let's get v1.1 into the fab first. We can't wait. Or is it already there? @gkasprow

**Comment by gkasprow** [comment]
@jordens do you need these FPGA pins for something else? Or just test point?
The SFP+ modules have I2C, while some SFP ones use these pins for bandwidth control.

**Comment by gkasprow** [comment]
I've just prepared new revisions for: Urukul, BNC, SMA, TTL tester, Sampler.
Now I'm working on Zotino, @marmeladapk will update Kasli ASAP.

**Comment by hartytp** [comment]
> The Silabs supply was not designed to be very low noise. We can add LDO.

If we can do this without adding too much additional complexity then I think it would be a good idea. Would it need an extra SMPS on Kasli though to produce the voltage that drives the LDO (or, can we use a 2V5 LDO driven from 3V3 for the Si chip?).

> I'd do that but also remove the clock fanout in favor of a single MMCX output form the Si5324 and then using Clocker or/and the backplane. Kasli is too dense and overloaded. The additional space would help with figuring out a good heatsinking solution, better SI, cleaner routing, higher yield, and lower manufacturing risk.

If it really does help then I'd be okay with this (although, it's definitely not something I'd like to see happen). However, before we do it I'd like @gkasprow to confirm that removing that clock buffer will really make a significant (at least noticeable) difference to any of the things you've said. e.g. I'm actually not convinced that removing that buffer will improve Kasli's SI or yield. Also, don't we still need a clock buffer to drive the BP clock lines as well as that MMCX?

**Comment by jordens** [comment]
@gkasprow AFAICT all modules use the rate select pins (RS0 and RS1) for bandwidth control. The MSA mandates I2C on MOD_DEF1/2.

@hartytp We have an unused diff output and one unused single ended on IC1 and an unused diff on IC2. Have the diff IC2 output go to the backplane, the single ended IC1 to an MMCX.
I am pretty sure the effects on routability, SI and cleanliness will be there. If the 7x MMCX array and the ADCLK948 are gone, that's a significant amount of space on all layers.

@marmeladapk And please finally remove the wire bridge W1. Even if it is DNP, it pierces through all layers and takes up lots of space.

**Comment by jordens** [comment]
@gkasprow I'd rather like to use these pins for:

1. RATE_SELECT1 (pin 9 of the SFP). It also gets rid of IC10{ABC}.
2. Reset of the I2C routers. If there lines shorted, on the I2C output ports (as there is on my board), there is no way to get out of that situation and reconfigure the I2C routers other than power cycling. That's bad.
3. Add test points and one more USER_LED.

**Comment by jordens** [comment]
You can put the EEPROM on the same I2C port as the SI5324. But I guess you have figured that out as well.

**Comment by jordens** [comment]
And I have changed my mind on the second power connector that is pointing towards the backplane. That can go (removed entirely). If someone wants to power internally, then just use the backplane connector, that EMI filter, and a connector on the backplane. In turn please populate the backplane connector.

**Comment by hartytp** [comment]
> I am pretty sure the effects on routability, SI and cleanliness will be there. If the 7x MMCX array and the ADCLK948 are gone, that's a significant amount of space on all layers.

Well, it's not a deal breaker for me, but I'd still prefer not to remove these unless @gkasprow and @marmeladapk really think it will help (they are the ones best placed to make that call). I would have no objection to cutting down to 4-5MMCX connectors if that helped (although, we discussed this before and, IIRC, @gkasprow and @marmeladapk said there was no point removing them since the cost is negligible and removing them doesn't really help with the design).

> And I have changed my mind on the second power connector that is pointing towards the backplane. That can go (removed entirely). If someone wants to power internally, then just use the backplane connector, that EMI filter, and a connector on the backplane. In turn please populate the backplane connector.

Great! I completely agree. Let's just make the BP adapter with a power jack on it.

**Comment by gkasprow** [comment]
@jordens true, I met some SFP modules that used these pines in other way, but they were custom CERN designs that also used both RD and TD as outputs :)

**Comment by gkasprow** [comment]
@jordens @marmeladapk On two kasli boards we noticed failed I2C switch. This may be due to ESD. The I2C are open drain with no protection diode. So we should add TVS to each EEM connector I2C lines.

**Comment by hartytp** [comment]
@jordens re the clock buffer: this is still a high value feature for us.
- I don't anticipate any noise issues with the 125MHz clock from the Si chip (assuming we add an LDO for it or at least some better PSU filtering). If we get close to matching the datasheet performance then it will as good as the AD9910 PPLs. Also, most of our Urukul applications aren't that noise sensitive anyway, so a few spurs won't hurt us.
- For Kasli DRTIO slaves, it's really convenient not to have to supply a clock via coax. This makes wiring much nicer. So, it's really nice to have a clock source output at the RTIO_COARSE frequency. That way, Kasli alone can supply everything  (power, DIO and clocks) needed to run EEMs.
- We generally have 100MHz references but not 125MHz/150MHz references, so it's really useful to have Kasli produce the 125MHz/150MHz we need to clock our DDSs from. It's also nice that we don't need to buy new reference oscillators for our DDSs if we change the RTIO frequency.
- We can use an external clock buffer such as clocker if we loose the fanout buffer on Kasli (so long as we keep at least an MMCX clock output). But then we have to source power for that (NB we plan to use all EEM connectors on Kasli, so there won't be one spare to power clocker in many cases). It also increases the cost of the overall system, takes up more rack space etc.
- The clock fanout isn't just a fanout but also a mux for the EEM clock fed to the backplane connector. If we loose the mux then we loose the ability to feed an external clock to the BP connector (we then *only* have the Si chip).
- While I understand your concerns, we have so far seen no evidence that the clock fanout chip on Kasli causes any SI issues etc on the prototype, so there is no actual evidence to back your concerns up. Also, @gkasprow and @marmeladapk -- the experts doing the design -- haven't yet agreed that the concerns are well founded.

**Comment by hartytp** [comment]
But, if you're really set on removing it then a possible compromise would be:
- Route the RTIO_coarse clock to the BP
- Keep a single MMCX on Kasli which also outputs the RTIO_coarse clock from the Si chip
- Put a 4x fanout + MMCXs on the Kasli BP adapter. Then we can use that for our applications.

**Comment by jordens** [comment]
@hartytp I am OK with exposing the clock at a few places (the backplane, an Kasli MMCX and the panel SMA). But I have the following observations with the 8x clock fanout:

* We definitely need additional space for an LDO for IC2, IC1.
* Kasli is extremely dense. I have been looking at the layout for the last couple weeks and I really don't like the fact that the density has forced Pawel to place all kinds of unrelated logic groups inside of each other. That makes debugging hard, makes reading the layout hard, and is just bad style. The fact that you can cram that much logic onto a board does not mean that you should do it. There are enormous downsides. It is also bad when changing things in the design as there are unnecessary knock-on changes and large parts have to be rerouted. The attitude of making future changes harder just because there is minute short term benefit is not a good one.
* The 7 or 8 coax and MMCX connectors make assembly more tricky and lead to a worsening of the cooling situation. It obstructs cooling (they run straight over the FPGA where the heatsink should be). Moving the 8x buffer of Kasli somewhere else would save a significant amount of power (and temperature) especially in the high density areas of IC1 and IC2.
* You can still do your 125 MHz fanout. You'll get that in three places: the SMA at the front, the backplane and the MMCX on Kasli. Just use Clocker. Otherwise, what purpose does Clocker serve since you would be getting 7 MMCX from Kasli? Using Clocker (or a fan-out on the backplane) makes wiring nicer as it reduces contention in the area where things are worst.
* You have to decide what's important for you. You say that the 125 MHz are "rare", but then you still want the option to not use them but feed the backplane with something else. This weakens both arguments.
* How would you use the MMCX input to the mux on Kasli if not from Clocker? That makes the mux on Kasli pretty useless as Clocker already gives you those outputs?
* Have the 2x8 fanout/mux on the backplane. You get the same mux options as before. And 8 outputs is sufficient to drive even setups with 8 Urukul/Mirny. That's unlikely to be standard.
* Sure there is evidence for SI issues. As Pawel and Greg stated, the routing is extremely dense and that lead to SI issues in the EEM connectors.

**Comment by hartytp** [comment]
> The 7 or 8 coax and MMCX connectors make assembly more tricky and lead to a worsening of the cooling situation. 

I would be totally happy with reducing that to 4.

> You have to decide what's important for you. You say that the 125 MHz are "rare", but then you still want the option to not use them but feed the backplane with something else. This weakens both arguments.

IIRC, you were the one who wanted that. I'd be fine with loosing the external input and only supporting backplane clocks from the Si chip. Just pointing out that we loose the muxing capacity if we remove the fanout. But, ...

> Have the 2x8 fanout/mux on the backplane. You get the same mux options as before. And 8 outputs is sufficient to drive even setups with 8 Urukul/Mirny. That's unlikely to be standard.


That's true. I hadn't considered that. The only downside of this is that there is no convenient way of feeding this mux via the front panel.

> Otherwise, what purpose does Clocker serve since you would be getting 7 MMCX from Kasli?

That was more targeting ultra-low noise cases. I'd prefer not to have to use if for simple fanout unless really necessary.

**Comment by hartytp** [comment]
But, having said all that, I would be okay with moving the mux + fanout to the BP adapter.

**Comment by marmeladapk** [comment]
@jordens And routing is very dense but near and under FPGA (eems routing was very tricky) not near clock circuits.

**Comment by jordens** [comment]
Making room to cool Kasli properly is high priority to me. A block of 8 EEM connectors, 8 ribbon cables, 8 MMCX connectors and 8 coax cables is an almost perfect wall. The opposition to adding 4 HP of space to Kasli is the wrong place to save space IMHO. Having racks filled in the front and in the back (like AMC/RTM) would be easier if HPs are at a premium and a the backside operation is apparently a small inconvenience. The EEM crates are really not deep right now. You can stuff boards from both sides in about 50 cm of crate depth. You can put Samplers on one side and Urukul on the other (at the same side as the RFPA which is much deeper). But please allow 8 HP for Kasli.

There is also the mux isolation issue. With the 75 dB spurs on Urukul with the capacitors placed to route the other input **and** the switch switched to the SMA, I'd expect something similar on the Kasli mux just by proximity. And it won't be easily curable there.

**Comment by hartytp** [comment]
> 8 coax cables is an almost perfect wall. 

Can reduce this to 4. But, even 8 coax cables aren't really an airflow problem compared to the ribbon cables.

> There is also the mux isolation issue. With the 75 dB spurs on Urukul with the capacitors placed to route the other input and the switch switched to the SMA, I'd expect something similar on the Kasli mux just by proximity. And it won't be easily curable there.

I have no issue with scrapping the mux and only keeping the fanout.

**Comment by hartytp** [comment]
Anyway, so long as we have a mux on Kasli or the BP adapter, I'm less fussed about where it is. So, will leave final call to @jordens and @marmeladapk. No more comments from me on this

**Comment by jordens** [comment]
@marmeladapk Ack. Thanks for the input. IMO any future layout changes would still be made easier if the 8x fanout and 7 MMCX were gone.
@hartytp What about the following: We scrap the idea of having the SMA connector output the RTIO clock and use that for input exclusively. They we add four MMCX on the four remaining outputs of the 4x fanout (two positive and two negative). The fact that there is a 180deg phase shift is not a problem IMHO. You can now go with a nice short pigtail from Kasli to its 8HP front panel if you want a RTIO clock output there. You can also go with 4 MMCX connectors to 4 Urukul/Mirny boards. You'd still have a differential output to the backplane (CLKOUT2 from IC2).

**Comment by hartytp** [comment]
@jordens that would be fine for me. I don't anticipate using the SMA clock output, since I imagine doing all clock distribution inside the rack. In the rare case that it's done one can add a pigtail/clocker (I don't mind doing that in this case since I don't think it will be very common, so it's okay if we don't make it quite as convenient).

**Comment by jordens** [comment]
@hartytp @marmeladapk Ok. Thanks. Then let me summarize:

* [ ] Remove IC3
* [ ] Add an LDO for the power supply of IC1 and IC2
* [ ] Remove the R98 path of CLK_OUT (IC1 to the front panel SMA)
* [ ] Connect four MMCX (J1-J4) to Q2, Q2B, Q3, Q3B of IC1
* [ ] Connect CLK_EXT_P/N to CKOUT2+- of IC2
* [ ] Remove the CLK_SEL logic of IC3
* [ ] Remove J19 and related components
* [ ] Remove J5-J7 and related components

**Comment by jordens** [comment]
@marmeladapk Why does R57 need to be 300R?

[UG470](https://www.xilinx.com/support/documentation/user_guides/ug470_7Series_Config.pdf#page=26):

> DONE has an internal pull-up resistor of
> approximately 10 kΩ. There is no setup/hold requirement
> for the DONE register. These changes, along with the
> DonePipe register software default, eliminate the need for
> the DriveDONE driver-option. External 330Ω resistor
> circuits are not required but can be used as they have been
> in previous generations.

**Comment by marmeladapk** [comment]
@jordens @hartytp Please tell me if you consider v1.1 to be a final version or do you expect v1.2?

> IMO any future layout changes would still be made easier if the 8x fanout and 7 MMCX were gone.

That part of the board was not dense and I had no trouble routing it. Removing anything is more work than DNPing it, unless you anticipate significant rework of clock recovery circuit schematics. So please consider if removing IC3 is still needed not on a premise that the board is dense, but on something else. If cost and power savings are worth it or if we're worried about cables obstructing cooling then we can DNP parts and move CLK_EXT to Q3 of IC1. 

> Add an LDO for the power supply of IC1 and IC2

We could supply IC1-3 with [TPS79601](http://www.ti.com/lit/ds/symlink/tps796.pdf) at 3,05 V from 3,3 V.

Alternatively, if we get rid of IC3 what do you think about supplying IC1 and IC2 with 2,5V from [LT3045EDD](http://datasheet.octopart.com/LT3045EDD%23PBF-Linear-Technology-datasheet-66874425.pdf)? Both chips can be powered with 2,5 V.

> Connect backplane clock CLK_EXT_P/N to CKOUT2+- of IC2

Do you want it to stay as LVPECL or change lines to LVDS?

> Why does R57 need to be 300R?

IDK, but with 2k2 Xilinx platform cable didn't recognise that the FPGA is programmed.

**Comment by hartytp** [comment]
> Please tell me if you consider v1.1 to be a final version or do you expect v1.2?

I currently don't have any more changes to suggest (although, I haven't played with these boards much yet).

> That part of the board was not dense and I had no trouble routing it. Removing anything is more work than DNPing it, unless you anticipate significant rework of clock recovery circuit schematics. So please consider if removing IC3 is still needed not on a premise that the board is dense, but on something else. If cost and power savings are worth it or if we're worried about cables obstructing cooling then we can DNP parts and move CLK_EXT to Q3 of IC1.

I'm not fussed either way.

> Do you want it to stay as LVPECL or change lines to LVDS?

IIRC, @gkasprow thought that LVPECL is better for driving cables etc than LVDS. I'd prefer to stick with it unless we need to change.

> We could supply IC1-3 with TPS79601 at 3,05 V from 3,3 V.
> Alternatively, if we get rid of IC3 what do you think about supplying IC1 and IC2 with 2,5V from LT3045EDD? Both chips can be powered with 2,5 V.

LGTM. Those regs are nice low noise regulators which should do the job. Anything alone those lines that meets the supply requirements of the chips is fine by me!

**Comment by marmeladapk** [comment]
@jordens Is there any reason other than visual that you brought back arc issue?

**Comment by jordens** [comment]
I hope that this will be the final revision of the v1.x series.

I am fine with DNPing instead of removing if it achieves the same.
Let me mark the changes that I definitely want to see in v1.1.

@marmeladapk Yes visual. I don't think the electrons care. But it looks amateurish, conveys a false sense of what is needed and important and seems to indicate that there is some hidden magic going. It's not a bug that prevents something from working but it is not good and something that I would like to see changed.

**Comment by jordens** [comment]
And if routing that clock network was as trouble-free as you say (which I find surprising), then re-routing it is even easier!
(That comment on the visual aspect of the layout does not refer to your skills which I admire. I hope that's clear).

**Comment by gkasprow** [comment]
@jordens existing resistor caused JTAG error that DONE pin did not go high. And once I changed it to 300R the problem was solved.

**Comment by jordens** [comment]
That's weird.

**Comment by gkasprow** [comment]
I had same issue with Spartan chips so didn't think too much about any other fix :)

**Comment by marmeladapk** [comment]
@jordens @hartytp So what's the consensus? Should I remove IC3 or should I just DNP J5-J7 and move CLK_EXT to Q3 of IC1?

And is there anyone else that should weigh in?

**Comment by jordens** [comment]
@marmeladapk I thought that was clear. Sorry if it wasn't. You can interpret the "Removes" from the items in the top post as "DNP" if you prefer. That's fine with me.

But **just** "DNP J5-J7" won't help. If we are doing any of this, I'd like the items marked **v1.1/rj** ticked off. That's more than DNPing a few components. It will involve re-routing some signals (e.g. driving J1-J4 from IC1 instead of IC3).

**Comment by marmeladapk** [comment]
@jordens If we drive J1-J4 from IC1 then and we DNP IC3 then we won't drive CLK_EXT with anything. I don't think that's intentional.

**Comment by jordens** [comment]
@marmeladapk 

> * (v1.1/rj) Connect backplane clock CLK_EXT_P/N to CKOUT2+- of IC2

**Comment by marmeladapk** [comment]
Ok, after some amount of discussion @jordens convinced me that shaving off 1 W and preventing possible isolation issues is worth it.

We plan to add this [heatsink](http://pl.farnell.com/wakefield-solutions/624-25abt4e/heat-sink-aluminium-21mm/dp/2802809) to BoM.

**Comment by marmeladapk** [comment]
> Remove the R98 path of CLK_OUT

I also removed R97 - no point in keeping it now.

**Comment by jordens** [comment]
Ack. Thanks!

**Comment by hartytp** [comment]
> @jordens @hartytp So what's the consensus? Should I remove IC3 or should I just DNP J5-J7 and move CLK_EXT to Q3 of IC1?

I'd vote for completely removing it rather than DNPing it. If you make the other changes that @jordens is suggesting then this component no longer makes any sense.

Plus IIRC by removing it, you can simplify things by powering the other clock components from 2V5.

**Comment by hartytp** [comment]
@marmeladapk Will the heat sink be glued down? How robust is that? Would it be better to use one that mounts robustly onto the PCB like on Sayma?

**Comment by marmeladapk** [comment]
@hartytp 

> Plus IIRC by removing it, you can simplify things by powering the other clock components from 2V5.

It doesn't really matter if they will be powered from 3 V or from 2,5 V.

> Would it be better to use one that mounts robustly onto the PCB like on Sayma?

Perhaps yes (@gkasprow ?) if we find one, but right now it would require significant rerouting near FPGA and I won't be able to finish it this week.

**Comment by hartytp** [comment]
> Perhaps yes (@gkasprow ?) if we find one, but right now it would require significant rerouting near FPGA and I won't be able to finish it this week.

Okay, do whatever you think is best.

But, just my two cents, I'd be a bit nervous about using one of those. There is so much cabling here that could knock a glued on heatsink off. I don't want to be in a situation where pulling a cable somewhere in the rack results in the heatsink falling off and then the FPGA breaks! So, I'd ideally like the board either to not rely on a heatsink or to have one that's sufficiently robust for the application.

**Comment by hartytp** [comment]
Anyway, all changes look good to me.

**Comment by gkasprow** [comment]
Glued heatsink is robust. I tried to remove one from Sayma and did not manage.
There are heatsink that are attached to the FPGA edge so don't need mounting holes. I used them for bigger ARTIX chips.

**Comment by hartytp** [comment]
@gkasprow if you're happy with it then it's fine by me. I've probably used too much cheap self-adhesive tape and not enough of decent glue :)

Use whatever approach you think will work best.

**Comment by gkasprow** [comment]
Such heatsink I sent requires quite a lot of clearance around FPGA.
If it does not fit, let's use glued one.

**Comment by hartytp** [comment]
@marmeladapk it's probably adding an inductor/ferrite between the SMPS and the new clock buffer LDO to make sure we kill high-frequency noise on that power rail.

**Comment by hartytp** [comment]
Also, if we're ever going to do it then let's tackle #317 while we're at it (should be pretty quick!).

**Comment by marmeladapk** [comment]
I also removed R131 and R133.

**Comment by hartytp** [comment]
One final comment (but not fussed if you don't want to make the change at this point):
- [ ] Should we float the SMA clock input by making that 0R resistor 100k?

My thinking here is that floating is generally what one wants for this. IIRC our only reason for not floating it originally was not wanting to have to fuss around with insulating washers. Since we're actually fitting an insulating washer in Kasli at the moment I can't see any reason not to float this input. Just requires changing one component value, so nice and easy to implement!

**Comment by hartytp** [comment]
FWIW, it probably would have been better to have floated the MMCX clock connectors on Kasli as well. If you can be bothered then let's do that.

However, if this is in any way hard to do (or if you just don't have time) then I don't mind you not bothering with it (it's not critical since these are generally only used for distribution over short distances within the rack, so ground loops shouldn't be too bad and, if this is a problem, one can use clocker)

**Comment by hartytp** [comment]
FWIW, one other thing we haven't looked at with Kasli is the robustness of the clock input. In particular, what happens when a clock is applied while the board is not powered (which always happens sooner or later!).

Quick calculation:
- Assume we apply a 2Vpp signal (this is the largest one typically uses) *from a generator with a 50R output impedance*
- Source voltage is thus 4Vpp=1.4Vrms, so short-circuit current is 28mA (actually a bit less that that, since current only flows once the voltage is greater than +-0.3V to forward bias the protection diodes)
- ADCLK chips can take 40mA current (presumably CW) so should be fine
- LTC6957-1 are only rated to 10mA input current, so may be damaged by applying a full-scale input when the board is unpowered (which is another argument against using them IMHO)
- I couldn't find a specification for those Si Labs chips, so who knows.

Anyway, this is probably fine but if we observe any damage then we could consider adding protection to the clock inputs (also applies to Urukul).

As discussed elsewhere, this is hard to do with diodes or protection resistors without degrading the performance of the clock chips, so adding one of the ADCLK buffers before the Si chip might be the easiest way of making the inputs more robust if there does turn out to be a problem...

**Comment by hartytp** [comment]
Other comments on that:
- If this does turn out to be a problem, adding diodes alone isn't necessarily enough, since the clock chip already has Schottky diode junctions at its inputs. So, we would probably need diodes and a current limiting resistor, placed between the diodes and the clock chip
- I don't want to add those components in, because it can be hard to do that without compromising the clock chip's noise performance. Doing this would probably need a bit of (time consuming) trial and error to determine things like maximum resistor value etc
- One thing I would be open to, if @gkasprow and @marmeladapk think it's worth it, would be to place a pad for back to back RF (fast switching, low capacitance) diodes (GND and the supply rail) and a current limiting resistor, but make the diodes DNP and start with the resistor 0R. That way if we do find a problem it can be fixed without changing the layout. We should probably also make sure to add a ferrite + capacitor filter between the diodes and supply rail (again, can be DNP for now) to avoid coupling noise onto the inputs. 
- Maybe we should have also done this for Urukul v1.2, but hopefully it's not needed in any case.

What do you think?

**Comment by sbourdeauducq** [comment]
> If this does turn out to be a problem, adding diodes alone isn't necessarily enough, since the clock chip already has Schottky diode junctions at its inputs. So, we would probably need diodes and a current limiting resistor, placed between the diodes and the clock chip

What would be the problem - excessive current through those diodes? Then what about adding a current-limiting resistor in series with the *power* pin of the clock chip (+decoupling of course), a diode, or a MOSFET-based switch?
(semi-related: https://hackaday.com/2011/05/20/using-an-avr-as-an-rfid-tag/)

**Comment by hartytp** [comment]
> I'm ok with floating the SMA connector if that's risk free.

The major risk I'm aware of here is that if one attaches a connector where the ground is at a significantly different potential (e.g. floating with ESD) then that now doesn't have a low-impedance path to circuit ground. There is a similar issue with using isolation transformers.

This is generally not an issue, but in a few cases it can lead to ESD-like damage of components (e.g. when ground potentials rapidly change relatively to each other). For the kind of lab work we're going to do I wouldn't anticipate this being an issue. Still worth being aware of.

> Why does no scope or spectrum analyzer manufacturer seem go that way e.g. on their reference inputs/outputs?

I can't give you a definitive answer on this (maybe Greg or Samer can) but a few comments:
- This is only an issue for measurements where low-level close-in phase noise matters (as, that's usually the only thing that ground loops on the clock inputs do). Most scopes and SAs aren't low enough noise for this to be a problem
- On scopes the grounding is typically a mess anyway as the inputs aren't floating (too high frequency for isolation amplifiers and need to be DC coupled), so it's not really worth worrying about ground loops in the reference input
- Things like our R&S FFT analyser (and also the SRS model) do support floated inputs for the signals at least to avoid this kind of ground loop problem.
- In general, one can always add an external inside-outside DC block so often manufacturers just rely on one doing that if needed. 

> I'd have preferred to have a non-transmission line galvanically isolating bal-un instead of that TLT (TR1). We should change that for Kasli v2.0. I overlooked it twice now...

IIRC the reason we decided not to do that was that the flux-cored baluns run out of steam rapidly above 1GHz. That means that you don't get a very nice square wave clock through them, so they're not great from a phase noise perspective.

For Kasli this may not be such an issue (since we don't care about phase noise too much), but I guess we figured that it was easiest/best to just do the same thing we're doing elsewhere (which would still be my preferred option).

**Comment by hartytp** [comment]
> What would be the problem - excessive current through those diodes? 

Exactly. But, only when a clock signal is applied while the clock chip is not powered.

When those things are properly biassed they're quite hard to kill. The diodes don't begin to conduct until the Vpp exceeds Vcc + 2*0.3V =3.1Vpp=1Vrms=+14dBm in our case. After that, the ADCLK chips can take 40mA through each diode. So, you're not likely to damage that with any standard signal generator.

When no supply is present, the positive-going part of the input signal sources current to the positive supply rail. That can be quite messy. But, in general, the supply rail charges up until enough circuitry powers up to provide a path to ground for this current. Usually to a volt or two. In rare cases, reverse biasing the LDO that normally supplies this rail can cause it to go into lockup (in which case, a diode is required across the LDO output).

> Then what about adding a current-limiting resistor in series with the power pin of the clock chip (+decoupling of course), a diode, or a MOSFET-based switch?
(semi-related: https://hackaday.com/2011/05/20/using-an-avr-as-an-rfid-tag/)

If you add resistance in series with the power rail, you also need to add it in series with the ground pin. These chips draw quite a lot of current, so to put enough resistance there to provide adequate protection can start to give a decent voltage drop across it, which is undesirable.

Also, FWIW, to get good close in phase noise, the power supply/ground needs to be low impedance all the way down to DC, so the decoupling capacitors aren't enough.

Not sure where you mean to put diodes? Between the connector and the clock chip? That's not a bad option, but:
- The clock chip already has Schottky diodes in it, so they will turn on at roughly the same time as the external diodes you add. So, it can be hard to ensure that the external diode actually draws the current away. To solve this, it's usual to add a resistor before the clock input to push more current through the external diode than the internal ones. The problem is that that external resistor can reduce the phase noise performance of the clock chips so some trial and error is needed to make sure that things actually work
- Since the signals we're applying are generally quite close to the supply rails, the external diodes can start to behave like non-linear capacitors to ground and cause SI issues. They can also couple noise from the supply rails to the very low noise clock inputs. Again, these issues can be overcome with good design, but it's easy to screw things up and kill the noise performance of the IC if one doesn't take due care and, again, some prototyping is generally needed.

Not sure what you mean by a MOSFET, but, again, there are the issues of ensuring that any components you add don't degrade the clock chip's performance, and thinking about how they operate when no supply is present.


----

This is all messy enough that it's probably not worth worrying about it for now and only fixing it if it turns out to be a problem in practice.

**Comment by hartytp** [comment]
One final thought about this: shall we populate the transformers on the clock inputs by default?

- they are really cheap, so they won't inflate the BOM cost noticeably. We'd only need to change a few population options, so it's really quick and easy to implement
- Standard oscillators like a Wenzel oscillator often put out +13dBm (2.85Vpp) or so. If we drive the clock chip single-ended then that's too much (they can only take something like 2Vpp). Adding the transformer reduces the voltage across each pin to 2.85Vpp/sqrt(2)=2Vpp (well, a bit less after 1dB or so of loss) so this becomes fine.
- Allowing bigger signals improves the phase noise performance when driven from a sinusoidal source
- For a similar reason, having the transformer makes these inputs more robust to having too much power applied.

I don't feel strongly about this, just raising it.

With all of this, @gkasprow and @marmeladapk just do whatever you think is best (or whatever you have time for).

**Comment by sbourdeauducq** [comment]
> Not sure what you mean by a MOSFET

In series with the clock chip's power, just like the diodes, without the drop-out voltage but needing more complex circuitry to turn it on/off with the board (like a synchronous rectifier). Then the rectified clock could only "power" the clock chip, which could reduce current to acceptable levels.

**Comment by hartytp** [comment]
That doesn't sound like a god idea to me at all.
- You still have to make sure that your FET switch functions properly when not biassed (taking into account all parasitic diode junctions)
- Unless you float the entire IC (which is not something I'd consider), you'll still have return paths via the output biasing resistors, digital pins etc. so that won't work.

Let's not overcomplicate (and potentially break) the design to try to solve something that we don't know is a problem -- hopefully, the ESD diodes on that Si chip are relatively beefy and can take the current that a standard +13dBm clock source can throw at them. If they can't then I still feel that diodes and a current limiting resistor on the input are the best way to protect the clock chip (although, that would probably need some prototyping to get right). Adding some pads for DNP/0R components was just intended as a quick way of ensuring that we can add extra protection in the future if needed without changing the layout, without actually adding them now.

**Comment by sbourdeauducq** [comment]
> You still have to make sure that your FET switch functions properly when not biassed (taking into account all parasitic diode junctions)

Sure; worst case use a small solid-state relay, that can't go wrong. Anyway, agreed we should not look too much into this unless it's a real problem.

**Comment by jordens** [comment]
I'd be fine with that as long as it doesn't delay anything and as long as we change them from TLTs to isolating bal-uns.

**Comment by hartytp** [comment]
> I'd be fine with that as long as it doesn't delay anything and as long as we change them from TLTs to isolating bal-uns.

With populating the baluns you mean?

Out of curiosity, why do you want to go for a flux-cored balun? For Kasli I'm not particularly fussed, but it does seem like a shame to use something that can't pass RF and I can't see an obvious advantage in terms of isolation (why do this galvanically rather than the current capacitive coupling)? Also, if you want to use the balun for isolation then you'll need to change the layout a bit IIRC since one half of the balun would need to be on the "ground island" with the connector...

**Comment by jordens** [comment]
Yes. For 100/125/150 MHz I'd like a DC isolated bal-un instead of trifilar one currently in the schematic. And yes, in that case make it a ground island. I like them because i would guess that breakdown voltage is higher than with capacitors. And AFAICT this (and not capacitors) is what people use in the industry to provide galvanic isolation. I expect them to have good reasons.
What is a "flux-cored" balun? AFAIK they all have flux in the core.

**Comment by marmeladapk** [comment]
@jordens Are you fine with some pin swapping? Either way there will be new pinout.

**Comment by jordens** [comment]
@marmeladapk depends a bit where. But generally yes.

**Comment by marmeladapk** [comment]
@jordens Only control lines for SFP.

**Comment by jordens** [comment]
Oh yes. Feel free. Same goes for LEDs.

**Comment by hartytp** [comment]
> What is a "flux-cored" balun? AFAIK they all have flux in the core.

Flux cored/magnetic cored/flux coupled is relatively [standard terminology](https://www.markimicrowave.com/Assets/appnotes/balun_basics_primer.pdf) IIRC. But, let's not get into a discussion about terminology here...

> I like them because i would guess that breakdown voltage is higher than with capacitors.

It certainly is, and AFAICT that's the reason they're usually preferred for lower frequency work.

FWIW, in our case I don't think the breakdown voltage is such an issue, as the clock source is unlikely to have a ground potential that differs from Kasli's by a significant amount. Also, we're capacitively coupling RF signals in various other places, so if this is a problem for Kasli then we need a rethink elsewhere and, if it's not an issue elsewhere then there's no point doing something different on Kasli.

As I said, other than it being nice to use the same transformer everywhere, the only real downside I can see of using the magnetic balun on Kasli is that it won't be able to pass nice square waves, which could make the phase noise performance a bit worse. But, that isn't going to be an issue as Kasli isn't ultra low noise anyway.

**Comment by hartytp** [comment]
> And AFAICT this (and not capacitors) is what people use in the industry to provide galvanic isolation. 

FWIW, capacitive inside-outside DC blocks are also quite common, particularly for microwave work (where they're the only option).

**Comment by jordens** [comment]
Oh sure, "flux-coupled". In the datasheet you cite, I can't find the term "flux-cored". I mean flux-coupled when I say "a DC-isolated transformer that is not a transmission line transformer".
Yes, we are capacitively coupling in several places on the board, just that we are not doing that with the ground/shield and we are not doing it yet between devices. That would parallel what industrial devices seem to be doing.
We won't have a noise problem with flux-coupled transformers on Kasli.

**Comment by hartytp** [comment]
Right on Kasli this is fine.

For urukul and Sayma it probably wouldn't be, which is why we have the pads for capacitivrly coupled grounds.

**Comment by a-shafir** [comment]
@hartytp it is a good question about "ground loops" in the labs. I think there is no real way to improve much on RF with coax/BNC in case if the connecting devices are not grounded nearly perfectly.
Breaking DC (and some low freq range) loop is good for preventing dangerous currents via PCB.
But IMHO need to be careful not to burn the sink resistor when it small or not to burn the following circuit with high voltages in case of some ESD or even DC dropped to the BNC/shield.

For mostly unbalanced i/o design like we have here the best approach is to have metal front panel and all the signals are referring to it so the back plane of the boards is "floating" to avoid currents trough the PCB(s). 
In case if the signals indeed can be affected by wide range device to device noises and parasitic currents need to switch somehow to balanced. I don't know if it sounds a crazy idea to use STP CAT-6 cables or similar approach?

Also for the clocks >10MHz will work 100pF or less capacitor for the signal and ground lines. It can easily stop DC, low freq noises and relatively high voltages as well . So unless we will put high RF on the input it will be safe and also will not affect the RF clock performance.

For the balun IMHO the main question how good the matching of the input (output) coax/BNC with the PCB diff circuit. For the critical nets it shall be simulated layout or tested with the network analyzer IMHO.

**Comment by dhslichter** [comment]
I think if you want sharp rise times with DC ground isolation and DC center pin isolation, inner/outer DC blocking caps seems like the most sensible way to go.  You don't care about frequencies below ~100 MHz, so finding appropriate component sizes should be fine.  The "standard"/non-transmission-line baluns usually tend to give up the ghost above ~400 MHz or so, which is why transmission line baluns were selected for this initially.

**Comment by marmeladapk** [comment]
Rerouting is almost done. Tommorow I'll improve power distribution and I'll run DC simulations.

**Comment by marmeladapk** [comment]
Simulations done. I must say, that removing back power connector made routing signals and power much easier. Upper 1/3rd of board was taken entirely by planes connecting to power connectors in middle layers.

![przechwytywanie](https://user-images.githubusercontent.com/24512604/35362837-48b163be-0167-11e8-9eec-4c9b8c98246c.PNG)
Is that note clear enough?

**Comment by marmeladapk** [comment]
@jordens @hartytp Just to be sure: clock fanout does not need to be length matched?

**Comment by marmeladapk** [comment]
I pushed RC version.

I can't remember what's our current stance on generated files, so didn't put PDF and BOM yet (seems that output folder dissapeared from Kasli, was it intentional?).

There's a bug with Altium, it didn't put 3d models for new parts, will solve that later.

Speaking of bugs, behold Altium generating teardrops:
![przechwytywanie](https://user-images.githubusercontent.com/24512604/35369393-03186e58-0188-11e8-8b83-380acda7ad00.PNG)
![przechwytywanie2](https://user-images.githubusercontent.com/24512604/35369400-04d86158-0188-11e8-9fdb-a7daf9bc8e24.PNG)

**Comment by jordens** [comment]
Thanks! The silkscreen markings are good. No length matching needed. I don't know who removed the files. But that's ok. Just tag a RC release and upload the files to the release.

**Comment by hartytp** [comment]
@marmeladapk Can you post the PDF to a RC release please so we can have a look over it? (Or, you can just attach it to a post here if that's easier).

**Comment by hartytp** [comment]
@gkasprow @marmeladapk One thing about the releases: please can you upload the PDF files directly to the releases as well/instead of putting them in the zip/7z file (as was done for many of the releases currently posted). It helps to have direct access to the shcematic PDFs without having to download and unzip a larger archive! Thanks.

**Comment by marmeladapk** [comment]
[KASLI.zip](https://github.com/m-labs/sinara/files/1663996/KASLI.zip)

**Comment by marmeladapk** [comment]
@hartytp @jordens  But how should we do it in the long run? Adding tags for each commit doesn't seem sustainable. I can upload PDF zip to comment commit.

**Comment by jordens** [comment]
@marmeladapk excellent. Very clean and tidy! I couldn't find anything I don't like on first glance. Will check more after lunch.
Commits and tags are diffferent. You should do commits when you change anything or when you want somebody else to look at the files in source form (using the viewer or the editor). When you want to do a release or release candidate, you'd tag the commit, create a release, and upload the files (not as a zip) to that release. That is  much better manageable than PDF zips in comments on issues or comments to commits. We have stable and semantic links to those files. That also makes space usage much more benign. We can delete/archive old releases or the data for releases and it is not a burden.

**Comment by marmeladapk** [comment]
@jordens Some arcs are still there. ;)

So I can assume that for incremental changes you'll use Altium Viewer?

**Comment by jordens** [comment]
[WIP]

* Would have been nice to do the I2C switch reset also (or predominantly) from the FTDI interface to the I2C tree. That's what's been bugging me so far (testing EEM I2C trees over USB, also for fabrication/QA). And the two switches don't need separate resets. Sorry, I wasn't clear on those two things.

**Comment by marmeladapk** [comment]
@jordens 

> I2C switch reset also (or predominantly) from the FTDI interface

Pyftdi (which, I assume, we want to use) doesn't support GPIO and I2C right now, so it would have to be connected to UART or 4th unused port. Also no point now in connecting the two resets together.

> This mode is mutually exclusive with advanced serial MPSSE features, such as I2C, SPI, JTAG, …
> If you need to use GPIO pins and MPSSE interface on the same port, you need to use the dedicated API. For now, this shared mode is only supported with the SPI API. [source](https://eblot.github.io/pyftdi/api/gpio.html)

**Comment by jordens** [comment]
Fast I2C over MPSSE is nice but not critical. It works fine at lower speeds with GPIO. I'd rather like to be able to reset the switches using GPIO. You can always switch to GPIO mode, do the reset, and then switch back to MPSSE mode. But anyway, this is minor. (1) I don't expect to need that anymore now that we have the TVS on there and (2) we can reset over JTAG/FPGA.

**Comment by marmeladapk** [comment]
> On the underside two of the I2C TVS are not populated. I guess that's Altium's fault.

That's a bug, 2V5 LDO and SATA also don't have 3d model.

> It would have been nice if the three USER_LEDs had been towards the front panel.

For me that's not a problem, but we would have to change panels. @gkasprow ?

**Comment by marmeladapk** [comment]
> You can always switch to GPIO mode, do the reset, and then switch back to MPSSE mode.

Good point.

**Comment by jordens** [comment]
* Yes. Don't bother moving the LEDs if the panels are already in the works.
* I really like the LT3045!
* If there is no major rerouting fallout it would be nice to have the I2C_SW1/2_RESET on CDBUS4. They really don't need to be separate.
* But do increase the pull up resistance on those reset pins to 10k if you need to pull them up to 3.3V or pull them up to 2.5V only: the FPGA is at 2.5 V.

**Comment by hartytp** [comment]
- [ ] @marmeladapk TR1 still needs to be populated and surrounding components changed.

**Comment by hartytp** [comment]
Overall though, looks great!

**Comment by hartytp** [comment]
In the power budget, are you double counting the power draw from the clock buffers, since this is listed under the 2V5 column and under the 3V3 column?

**Comment by jordens** [comment]
Ok. I am done reviewing. Let me know what you think about the four items marked (**!**) above. Great work!

**Comment by hartytp** [comment]
@marmeladapk Where are the PCB mount clips that hold the heatsink onto the board?

**Comment by jordens** [comment]
@hartytp IIRC we said there is no space. Needs to be a glued heat sink.

**Comment by hartytp** [comment]
@jordens you're right, I'd forgotten about that. Thanks for reminding me.

**Comment by marmeladapk** [comment]
@jordens 

> I2C_SW1/2_RESET on CDBUS4
> increase the pull up resistance on those I2C_SW1/2_RESET pins to 10k

And disconnect it from FPGA or keep it? If we're keeping we should add resistor in series. Connecting RESET to CDBUS4 won't be a problem.

> Good work keeping the GTP diff pairs further apart!

I didn't touch them ;) I only removed lines in-between them and routed them where unfiltered P12V0 ran in v1.0.

> put the MMCX shield grounding component groups like C43/R29/C44 closer to the center conductor capacitor

ACK.

> rotate the (SATA) connector

ACK.

@hartytp 

> TR1 still needs to be populated and surrounding components changed.

ACK. I didn't see a clear conclusion from your discussion, that's why I didn't change it.

> In the power budget, are you double counting the power draw from the clock buffers

No, total power draw is only calculated from 3V3, 2V5, 1V5 and 1V0.

**Comment by jordens** [comment]
@marmeladapk 
* I am fine with having I2C_SW1/2_RESET go only to the FTDI chip and not to the FPGA. Situations where the FPGA needs to talk I2C **and** the I2C tree is broken must be fixed by some other means. That also removes the issue of pull-up strength on those signals. And merging them is fine as well.
* On the SATA connector. Really don't bother rtating it if that poses any hassle. If you do rotate it, I think the best rotation would be (90deg counter clockwise)[https://pactech-inc.com/wp-content/uploads/2014/06/SATA2K-XXL-Amphenol-SATA-Cable-Straight-with-Latch-to-Left-Angle-003.jpg] with the notch pointing down.
* I am for populating the bal-un as well. I like bal-uns. Let's grant us that luxury.

**Comment by jordens** [comment]
For the SATA connector it doesn't matter wither it is 90 degree CCW or 90 degree CW. Both seem to be [equally present](https://www.digikey.de/products/en/cable-assemblies/pluggable-cables/460?k=sata&k=&pkeyword=sata&pv28=2400&pv28=2401&FV=16680012%2Cffe001cc&mnonly=0&ColumnSort=-1000009&page=1&stock=1&quantity=0&ptm=0&fid=0&pageSize=25).

**Comment by marmeladapk** [comment]
@jordens Rotating clockwise is easier. 

I need to add an additional buffer between FTDI and I2C switch for this reset signal.

**Comment by marmeladapk** [comment]
> EMI ground via stitching at the PCB edges?

Not sure if this is necessary because all high-speed signals are routed in the middle of the board (X/Y/Z) (except maybe SATA), however I have no experience with so @gkasprow comment is needed.

**Comment by gkasprow** [comment]
It does not matter if traces are routed in the middle or not. The open edge of the board forms slot antenna and irradiates EMI. That's why we place vias at the board edges that short all ground planes and form some kind of Faraday cage for internal layers.

**Comment by marmeladapk** [comment]
@jordens 
![przechwytywanie3](https://user-images.githubusercontent.com/24512604/35409646-25b2635c-0213-11e8-8ee9-dafb9f357ec0.PNG)

Rotating SATA connector is harder than I thought. It's both too close to the edge and to the JTAG connector. Moving JTAG down would require rerouting GTP pairs (and many more signals) and is not optimal. Rotating it would create problems with P12V0 distribution and also requires rerouting. Since you said to pass if it's troublesome, I'm going to defer it to next revision.

**Comment by marmeladapk** [comment]
I found better [heatsink](https://octopart.com/ick+bga+23x23x10-fischer+elektronik-20057069)

http://www.fischerelektronik.de/web_fischer/en_GB/heatsinks/B02/Heatsinks%20for%20BGAs/PR/ICKBGA23x23x10_/$productCard/parameters/index.xhtml

It has dimension of FPGA and is self-adhesive.
![przechwytywanie5](https://user-images.githubusercontent.com/24512604/35413577-2316374c-0220-11e8-8ff8-0cf0ab928b37.PNG)

**Comment by hartytp** [comment]
So long as the glue is tough enough to survive lab usage (e.g. being nocked by all those ribbon cables) that looks perfect!

**Comment by jordens** [comment]
@marmeladapk all ACK. Then we are good to go.

**Comment by marmeladapk** [comment]
I pushed RC2. Please check if shielding vias aren't too close to board edge.

**Comment by hartytp** [comment]
@marmeladapk Thanks for uploading the R2C release along with the change notes. That's really helpful and exactly how I think this should be done!

All looks great on the schematic now, and I'd be happy to close this and send it off for manufacture. Good job.

One thing though, I'd really like to close all outstanding Kasli issues, including #317. So, could you either add that info to the schematic and close the issue, or let's agree that this info isn't important and close the issue.

**Comment by jordens** [comment]
For the stitching vias to have any effect, they need to connect the top and bottom ground pours (as Greg explained) and they need to be closely spaced (~four times as dense as you have them now). I don't think you are doing that. But @gkasprow knows best what they should look like and whether they are needed.

* I would not short pin 4 (B2) of IC10 to ground. Even though it's low, it's still an output.
* (my fault, sorry, but this is a bug) Connect IC10:A1 to one of the unused BDBUS lines (BDBUS7 if you can) and **not** to CDBUS4. I2C is on the B bus now and that's where the reset should be as well. I looked at the wrong schematic.
* I am not overly excited to see the I2C bus and the UART scattered over the banks on the FPGA. The various SFP control lines would have been better candidates for scattering. But I think this is OK for now (@sbourdeauducq could you confirm?)
* IIRC we also said that if we use a balun as input, we want a DC-isolated one. @hartytp What about the [TC2-1TX+](https://ww2.minicircuits.com/pdfs/TC2-1TX+.pdf)? Almost the exact same footprint, just rotated and with one pin less. Nice for pick-and place (same top hat). Works for all frequencies we might want to put into the Si5324 that I can think of right now (3-300 MHz). There are others with lower and higher 3dB frequencies but this one matches the 5-100 MHz (plus 3rd harmonic for steeper edges from square sources) and footprint best. Maybe we want to DNP C13, C9 as well then.

**Comment by hartytp** [comment]
> IIRC we also said that if we use a balun as input, we want a DC-isolated one

Well, I'm still not keen on this. As discussed before, I don't think that the high isolation voltage afforded by the DC-isolated balun is useful, and it will mangle the rising edges of any clocks we feed through, as it only passes the fundamental of a 150MHz clock. This will make the clock's noise immunity much worse, as we can't feed in signals with rising edges.

I also object to using a different balun/isolation method here to everywhere else in Sinara. It's one more datasheet for users to have to look at, and it gives the user the impression that there is somethign special about this input. IMHO, unless there is a real need to do something different here, it would be much better to just stick with the solution that we've used everywhere else and tested carefully. (Not to mention that sticking with the current solution avoids having to do extra work).

But, that said, it's your board and your call...

---

Assuming the TC2-1TX+ is pin-compatible with the current balun, it looks okay to me (apart from the crap frequency cut-off).

**Comment by hartytp** [comment]
@jordens Also, if you want to change the balun to improve the isolation voltage then other changes are needed as well:
- DNP the 100k resistor and capacitors on the input (if the caps are still there then they limit the breakdown voltage)
- Add a ground island around the balun. This ground island must not be connected to the main pcb ground, even with capacitors (I don't really like from an EMI perspective, given that there are high-frequency signals around that are far outside the BW of your balun)
- generally, review all components around the balun and input to make sure it does what you want and works

Do we really want to go to that effort at this late stage in the design? It's not so hard, but there are a few ways of screwing things up, so I'd prefer to stick with our current balun situation, which we have tested elsewhere (e.g. clocker) and know should work...

**Comment by jordens** [comment]
@hartytp 
(1) The 5, 10 or 100 MHz outputs from reference source we would be using is sinusoidal. There is no point in accommodating for what we won't use.
(2) In cases where Kasli is DRTIO master, I don't see anyone inputting a clock other than sinusoidal or other than 5/10/100 MHz. As you say, 150 MHz and 125 MHz are rare.
(3) The previous balun has a crap lower frequency cut of as well. 5 MHz is not such an uncommon reference frequency. And loosing SNR there is so much worse than at 100 MHz because of the lower frequency.
(4) I feel zero obligation to follow the Sayma clock input designs. It's an illusion that we are doing that (no squaring up chip, different PLL, different bandwidth, different application). The Urukul output stages already use a different Balun (I am fine with copying that design. It goes to higher frequencies but it's a different footprint). The Sayma design is neither known to be a good choice -- certainly no testing at all at low frequencies or with a narrow band PLL -- nor do we have the data to support the claim that the TLT is better noise-wise.
(5) Remember that the typical PLL bandwidth of the Si5324 is 5-10 Hz. I have my doubts that you will see any relevant phase noise difference when using 150 MHz. At 5-10 Hz bandwidth there will be other effects dominating.

**Comment by jordens** [comment]
Clocker is tested already?

**Comment by hartytp** [comment]
> Clocker is tested already?

I haven't tested the version @gkasprow built yet, but we have something essentially identical, which we've been using for quite a while and have tested very carefully (this is what was used for a lot of the phase noise plots I've posted so far).

Anyway, I don't feel strongly enough to argue with you about this. Since you're the one pushing for it, just make sure you review all changes carefully it doesn't break anything...

**Comment by hartytp** [comment]
> It goes to higher frequencies but it's a different footprint

Oh, and I don't care too much about changing the footprint. I do not want to support multiple population versions for this, so let's just pick a balun and get it working.

**Comment by jordens** [comment]
On the breakdown I would keep the capacitors on the shield as they are now and also don't make an island. I am not trying to protect against discharge into the SMA shield, the case, the panel, SFP cage, 12V outer barrel. I am more worried about the signal path into the Si5324.

**Comment by jordens** [comment]
@hartytp You can also populate the [TC2-72T+](https://ww2.minicircuits.com/pdfs/TC2-72T+.pdf). With 700 MHz that's more than the required 4V/ns bandwidth the ADCLK would want. But since the is not the ADCLK948 and we don't have any data about how the Si5324 reacts to diffferent slew rates and how it's 5 Hz PLL sees that, I don't see the point.

Ok. @marmeladapk Then please change the

**Comment by marmeladapk** [comment]
SN74AVC2T245 datasheet says to "Tie any unused input and output ports directly to ground."

@gkasprow recommended me 10 mm spacing. I'll move vias closer to the middle to also connect with top and bottom ground polygons.

Other changes will have to wait till Sunday at least. I need to catch up with few things on my uni. Try not to change the specification after Sunday, since I feel like I'm trying to hit a moving target. ;)

@jordens Could you update top post with these last things you and @hartytp discussed? So it's clear which changes do you want to be implemented.

**Comment by marmeladapk** [comment]
> I'd really like to close all outstanding Kasli issues, including #317

@hartytp I'll complete it, but after board is finalised.

**Comment by jordens** [comment]
@marmeladapk all ACK.

**Comment by hartytp** [comment]
> TR1 to TC2-1TX+ for my boards

Let's do this for all boards then. I want to minimize the number of population variants floating around for this hardware.

**Comment by jordens** [comment]
@marmeladapk I saw your note on the XADC issues. Are you using vivado with JTAG to access the XADC directly or do you have gateware that you are using to access it?

**Comment by gkasprow** [comment]
we use Vivado, I also observed such effect. But after some time it started
working.

>

**Comment by sbourdeauducq** [comment]
> I am not overly excited to see the I2C bus and the UART scattered over the banks on the FPGA. The various SFP control lines would have been better candidates for scattering. But I think this is OK for now (@sbourdeauducq could you confirm?)

I2C and UART are low-speed signals, and they are using the most regular FPGA IO (no clock-capable pins etc.). So there shouldn't be an issue with scattering them.

**Comment by jordens** [comment]
@marmeladapk "The XADC also has an on-chip reference option which is selected by connecting VREFP and VREFN to ADCGND": replace C49 with a 0R.

**Comment by marmeladapk** [comment]
@jordens I connected it without a jumper. C49 was in 0201 case, our jumpers are 0402, there's no way I could fit that under FPGA.

**Comment by jordens** [comment]
@marmeladapk thanks.

**Comment by marmeladapk** [comment]
@jordens If I just flip TC2-1TX+ by 180 deg. then output will be swapped (_P line with _N line). Do we care? I can connect CLK_IN to pin 4 if we want to avoid that.

![sch](https://user-images.githubusercontent.com/24512604/35573505-9d18e7f0-05d7-11e8-98e9-c48b3e64f718.PNG)
![pcb](https://user-images.githubusercontent.com/24512604/35573507-9f8592a4-05d7-11e8-8da2-f1a68cf03227.PNG)
(up - current balun, down, TC2-1TX+ rotaded 180 deg.)

**Comment by jordens** [comment]
@marmeladapk Correct (well observed!). The swapped polarity doesn't hurt me but if you want (and for the perfectionists inside us) swapping the connection on the primary (6-4) is nicer.

**Comment by marmeladapk** [comment]
I finished, but I'll upload documentation from my home computer, as here I've got some issues with generating it (new Altium... ).

**Comment by marmeladapk** [comment]
@jordens @hartytp I released rc3. Unless you have any objection I'd like to send it to the manufacturer.

**Comment by jordens** [comment]
@marmeladapk thanks! Give me an hour to check it.

**Comment by hartytp** [comment]
@marmeladapk looks beautiful. If I had to nit pick, I'd probably suggest adding a PN for that clock balun on the schematic. Otherwise, no comments here apart from send it off! Great work!

**Comment by jordens** [comment]
@marmeladapk Good to go! Send it!

**Comment by hartytp** [comment]
Whooo!

**Comment by marmeladapk** [comment]
@hartytp I'll fix schematics in the final version.

**Comment by marmeladapk** [comment]
Note to self: I unmasked thermal vias.

**Comment by marmeladapk** [comment]
As a part of one of my projects on uni I wrote code that controls all parameters of Si5324. It abstracts all the registers stuff, allows to set desired input clock and output frequency (but only on one output). It finds right divider values so you don't need to use DSPLLsim for this but it does so using brute force method. In all cases I tested it found right values in <0.5 second (even with odd frequencies like 192.3 MHz). Do you want me to upload it for later use?

**Comment by gkasprow** [comment]
@marmeladapk ask @jordens

**Comment by jordens** [comment]
Integrate it into the code I posted before.

**Comment by marmeladapk** [comment]
@jordens V1.1 arrived yesterday. Unfortunately pyftdi resets all other outputs when using I2C, so we have to fall back to bitbang. :/

**Comment by jordens** [comment]
@marmeladapk You mean it messes with the other channels? Yes. I have seen that too. 
Or do you mean it resets the other pins on the second channel? That should and can be fixed in pyftdi as well.
In both cases, could you file a bug with pyftdi if there isn't one yet?

**Comment by jordens** [comment]
Anyway, bitbanging I2C is just fine.

**Comment by marmeladapk** [comment]
@jordens @hartytp Unless anyone has any objections to schematics, I'd like to publish v1.1 and upload board files.

**Comment by hartytp** [comment]
Go for it.

**Comment by jordens** [comment]
:+1:

**Comment by marmeladapk** [comment]
v1.1 is released, so I'll go ahead and close this.
